duckduckgo/macos-browser
# Internal testing channel

Task/Issue URL: https://app.asana.com/0/0/1203756847210448/f
Tech Design URL: https://app.asana.com/0/481882893211075/1203837424896950/f

**Description**:
Internal testing channel is a release channel that receives updates sooner then they are available to public. DuckDuckGo team members are automatically added to the internal channel once they load "use-login.duckduckgo.com" successfully. Once app build is marked as internal, Sparkle updater is able to install builds marked as internal in the appcast file.

**Steps to test this PR**:
1. Comment following lines so debug build behaves like release build
AppIconChanger.swift
```
//#if DEBUG || REVIEW
//         return
// #else
         let icon: NSImage?
         if isInternalChannel {
             icon = NSImage(named: Icons.internalChannelIcon.rawValue)
         } else {
             icon = nil
         }

         NSApplication.shared.applicationIconImage = icon
// #endif
```
InternalUserDecider.swift
```
//#if DEBUG || REVIEW
//        if AppDelegate.isRunningTests {
//            isInternalUser = (try? store.load()) ?? false
//        } else {
//            isInternalUser = true
//        }
//#else
        isInternalUser = (try? store.load()) ?? false
//#endif
```

2. Clean browsing data using fire button (make sure duckduckgo.com is not fireproofed)

3. Visit bunch of websites including duckduckgo.com and make sure the icon remains blue

4. Login to Asana

5. Visit http://use-login.duckduckgo.com

6. Make sure the icon changed to "TEST" icon

7. Restart the app and make sure the icon is still changed

8. Add a breakpoint to `allowedChannels(for updater: SPUUpdater) -> Set<String>` 
9. Click on "Check for Updates..."
10. Make sure the method returns a set with "internal-channel" string



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
